modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+      ),
+    );
+
+    $rows[] = $row;
+  }
+
+  $output = theme(
+    'table',
+    array(
+      'header' => $header,
+      'rows' => $rows,
+      'attributes' => array(
+        'id' => 'ding_nodelist-filters-table',
+      ),
+    )
+  );
+  $output .= drupal_render_children($form);
+  return $output;
+}
+
+/**
+ * Helper theming function.
+ */
+function theme__ding_nodelist_links_table($variables) {
+  $form = $variables['form'];
+  $rows = array();
+  $header = array(t('Text'), t('URL'));
+
+  foreach (element_children($form) as $key) {
+    // Build the table row.
+    $row = array(
+      'data' => array(
+        drupal_render($form[$key]['text']),
+        drupal_render($form[$key]['links']),
+      ),
+    );
+
+    $rows[] = $row;
+  }
+
+  $output = theme(
+    'table',
+    array(
+      'header' => $header,
+      'rows' => $rows,
+      'attributes' => array(
+        'id' => 'ding_nodelist-links-table',
+      ),
+    )
+  );
+  $output .= drupal_render_children($form);
+  return $output;
+}
+
+/**
+ * Theme function to render the backend list of templates.
+ */
+function theme__ding_nodelist_templates_table($variables) {
+  $header = array(
+    t('Machine name'),
+    t('Title'),
+    t('Content type'),
+    t('List type'),
+    t('Visible'),
+  );
+  $rows = array();
+  $form = $variables['form'];
+  foreach (element_children($form) as $key) {
+    $rows[] = array(
+      $form[$key]['template_name']['#value'],
+      drupal_render($form[$key]['title']),
+      drupal_render($form[$key]['content_type']),
+      drupal_render($form[$key]['widget']),
+      drupal_render($form[$key]['visible']),
+    );
+  }
+
+  return theme('table', array('header' => $header, 'rows' => $rows)) . drupal_render_children($form);
+}
+
+/**
+ * Theme function to render backend lisy of widgets that require text trim.
+ */
+function theme__ding_nodelist_trim_text_table($variables) {
+  $header = array(
+    t('Widget'),
+    t('Title length'),
+    t('Teaser length'),
+  );
+
+  $rows = array();
+  $form = $variables['form'];
+  foreach (element_children($form) as $key) {
+    $rows[] = array(
+      $form[$key]['widget']['#value'],
+      drupal_render($form[$key]['title_trim']),
+      drupal_render($form[$key]['teaser_trim']),
+    );
+  }
+  return theme('table', array('header' => $header, 'rows' => $rows)) . drupal_render_children($form);
+}
+
+/**
+ * Ctools admin view pane's title.
+ */
+function ding_nodelist_content_type_admin_title($subtype, $conf, $context = NULL) {
+  if (!empty($conf['override_title_text'])) {
+    $output = t('Nodelist') . ': ' . filter_xss_admin($conf['override_title_text']);
+  }
+  else {
+    $output = t('Nodelist (untitled)');
+  }
+  return $output;
+}
+
+/**
+ * Callback to provide admin info (the preview in panels when building a panel).
+ */
+function ding_nodelist_content_type_admin_info($subtype, $conf, $context = NULL) {
+  // @todo: show details about nodelist configuration here.
```

Comment:
Having nodelist information in the admin interface seems important. Especially given the complexity of the configuration, the fact that one page could have multiple panes etc.